src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -3230,6 +3230,25 @@ public string Parameters
             /// </summary>
             public void Initialize(IEventSource eventSource)
             {
+                // The concrete type we get should always be our internal
+                // implementation and up-to-date, but we need to meet the
+                // external contract so can't specify that for the
+                // argument.
+
+                IEventSource4 eventSource4 = (IEventSource4)eventSource;
```

Comment:
Isn't IEventSource public? Can't someone have made their own IEventSource that doesn't implement IEventSource4?